From 6704c0590db779f90e0dad304ba9d0c3cf1bc72a Mon Sep 17 00:00:00 2001
From: Sandy Huang <hjc@rock-chips.com>
Date: Wed, 6 Nov 2019 15:12:03 +0800
Subject: [PATCH] drm/rockchip: vop: fix plane number calc error

before this commit, the plane number to dmc is error, this
will lead to DDR freq is too low and lead to display splash.

Change-Id: Iaa285b36d41b6e86c97260ac83c5d9a21ef62a05
Signed-off-by: Sandy Huang <hjc@rock-chips.com>
---
 drivers/gpu/drm/rockchip/rockchip_drm_vop.c | 9 +++------
 1 file changed, 3 insertions(+), 6 deletions(-)

diff --git a/drivers/gpu/drm/rockchip/rockchip_drm_vop.c b/drivers/gpu/drm/rockchip/rockchip_drm_vop.c
index 688296788ed7..7b484d23e447 100644
--- a/drivers/gpu/drm/rockchip/rockchip_drm_vop.c
+++ b/drivers/gpu/drm/rockchip/rockchip_drm_vop.c
@@ -2481,6 +2481,7 @@ static size_t vop_crtc_bandwidth(struct drm_crtc *crtc,
 	struct drm_plane *plane;
 	u64 bandwidth;
 	int cnt = 0, plane_num = 0;
+	struct drm_atomic_state *state = crtc_state->state;
 #if defined(CONFIG_ROCKCHIP_DRM_DEBUG)
 	struct vop_dump_list *pos, *n;
 #endif
@@ -2503,12 +2504,8 @@ static size_t vop_crtc_bandwidth(struct drm_crtc *crtc,
 	}
 #endif
 
-	drm_atomic_crtc_state_for_each_plane(plane, crtc_state) {
-		pstate = plane->state;
-		if (pstate->crtc != crtc || !pstate->fb)
-			continue;
+	drm_atomic_crtc_state_for_each_plane(plane, crtc_state)
 		plane_num++;
-	}
 
 	if (plane_num_total)
 		*plane_num_total += plane_num;
@@ -2518,7 +2515,7 @@ static size_t vop_crtc_bandwidth(struct drm_crtc *crtc,
 		return -ENOMEM;
 
 	drm_atomic_crtc_state_for_each_plane(plane, crtc_state) {
-		pstate = plane->state;
+		pstate = drm_atomic_get_existing_plane_state(state, plane);
 		if (pstate->crtc != crtc || !pstate->fb)
 			continue;
 
-- 
2.35.3

